e2e/onboarding.e2e.spec.js

```diff
@@ -0,0 +1,232 @@
+import {expect, element, by, waitFor} from 'detox';
+import {isIOS, waitForElement} from './utils';
+import en from '../src/assets/lang/en.json';
+
+// Test timeout for long-running tests (default value is 120 seconds).
+const longRunningTestTimeout = 240 * 1000;
+
+describe('user onboarding flow', () => {
+  describe('when the user opens the app with no data stored', () => {
+    it(
+      'should show the user the relevant data protection and metric notices',
+      async () => {
+        // Confirm you're over 16 and continue past the get started view.
+        await element(by.text(en.ageRequirement.over)).tap();
+        await element(by.text(en.onboarding.intro.action)).tap();
+
+        // Ensure the data protection view is displayed.
+        await expect(element(by.text(en.yourData.title))).toBeVisible();
+
+        // Tap on the data protection information notice and ensure it displays.
+        await element(by.text(en.dataProtectionPolicy.link)).tap();
+        await expect(element(by.text(en.settings.privacyPolicy))).toBeVisible();
+
+        // Go back to the previous view.
+        await element(by.text(en.navbar.back)).tap();
+
+        // Scroll until the contiune button is visible and tap it.
+        const continueButtonEl = element(by.text(en.yourData.continue));
+        const scrollViewId = by.id('your-data');
+        await waitFor(continueButtonEl)
+          .toBeVisible()
+          .whileElement(scrollViewId)
+          .scroll(100, 'down');
+        await continueButtonEl.tap();
+
+        // The try/catch block tries to solve an edge case on Android:
+        // 1. The emulator is not running.
+        // 2. Detox runs the emulator and then the test.
+        // 3. Tapping the continue button calls, among other things, the Google SafetyNet Attestation API.
+        // 4. The attestation call fails with error code "7".
+        // 5. The app remains in Your Data screen and displays an error toast.
+        // Note: the attestation call does not fail if the emulator is already running before Detox runs the test.
+        // If the app doesn't move to the App Metrics screen, the test taps the continue button again.
+        // This time the attestation call should succeed.
+        const appMetricsTitle = by.text(en.appUsage.title);
+        try {
+          // Ensure the app metrics view is displayed.
+          await waitForElement(appMetricsTitle, 'toBeVisible', 90);
+        } catch (errNotFound) {
+          await waitFor(continueButtonEl)
+            .toBeVisible()
+            .whileElement(scrollViewId)
+            .scroll(100, 'down');
+          await continueButtonEl.tap();
+          await waitForElement(appMetricsTitle, 'toBeVisible');
+        }
```

Comment:
This `try/catch` block solves an issue happening with the Android emulator.

If the Android emulator is not running before Detox runs the test, some tests fail when the user clicks the Continue button in "Your Data" screen.
This is happening because the button triggers the `register` call (`src/services/api/index.ts`), which calls the `verify` function (`src/services/api/index.ts`). Here, the [Google SafetyNet Attestation call](https://github.com/covidgreen/covid-green-app/blob/current/src/services/api/index.ts#L27) fails with a cryptic `[Error: 7: ]`. At this point, the app remains in "Your Data" screen and displays a network error toast. The test fails because it expects to find the next screen title ("App Metrics"), but it doesn't.

To detect this error, I had to increase Detox expectation waiting time from 10 to 90 seconds because the attestation call takes about 60 seconds to fail on my machine. 

I was not able to find a fix for the `Error 7`problem, I've checked the `react-native-google-safetynet` repo, and there is an open issue about this type of error (https://github.com/rajivshah3/react-native-google-safetynet/issues/386). So let's wait to be clarified/fixed by the owner of the library.

I've put in place a workaround to avoid test failing in this particular case: I wait 90 seconds to see if the "App Metrics" title appears; if it doesn't, Detox throws an exception and the app is still in "Your Data" screen. I catch this exception, I click the Continue button and check the existence of the "App Metrics" title again. In my tests, the second Attestation call always works, so the test can continue.